tuning/tune-wmma.sh

```diff
@@ -1,8 +1,82 @@
 #!/usr/bin/env bash
 set -Eeuo pipefail
 
-cd "$( dirname "${BASH_SOURCE[0]}" )"
+GPU_CLOCK=915
+MEM_CLOCK=5001
+
+usage()
+{
+    cat <<EOF >&2
+Usage: $0 [OPTIONS]
+
+Tune WMMA Parameters.
+
+Options:
+-h, --help                 Show this help.
+-gc, --gpu-clock speed     Change the frequency the GPU core clock is locked to
+                           before benchmarking, in MHz (default 915 MHz).
+-mc, --memory-clock speed  Change the frequency the GPU memory clock is locked to
+                           before benchmarking, in MHz (default 5001 MHz).
+EOF
+}
+
+positional=()
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -h|--help)
+            usage; exit 0
+            ;;
+        -gc|--gpu-clock)
+            shift
+            GPU_CLOCK=$1
+            shift
+            ;;
+        -mc|--memory-clock)
+            shift
+            MEM_CLOCK=$1
+            shift
+            ;;
+        -*)
+            echo "Unknown command-line option '$1'."
+            echo "Try '$0 --help' for more information."
+            exit 1
+            ;;
+        *)
+            positional+=("$1")
+            shift
+            ;;
+    esac
+done
+set -- "${positional[@]}"
+
+if [[ $# -ne 0 ]]; then
+    echo "Expected 0 positional arguments, but got $#."
+    echo "Try '$0 --help' for more information."
+    exit 1
+fi
 
+echo "Locking GPU clock speeds to $GPU_CLOCK MHz (GPU) / $MEM_CLOCK MHz (Mem)..."
+
+if ! nvidia-smi --query-supported-clocks=graphics,memory --format=csv | grep -F "$GPU_CLOCK MHz, $MEM_CLOCK MHz"; then
+    echo "Unsupported combination of clock speeds!"
+    exit 1
+fi
+
+# Prompt for sudo
+sudo -v &>/dev/null
+
+# Sudo keep-alive
+while true; do
+    sleep 300
+    sudo -n true
+    kill -0 "$$" || exit
+done &> /dev/null &
```

Comment:
Ha, cute! Maybe could be more reliable (e.g. when sudo is configured always to prompt again) to fork off a privileged process that waits for the parent though.